### ISSUE TITLE
@JsonUnwrapped on Optional value does not propagate prefix to schema

### ISSUE THREAD

**Issue by osi** [FasterXML/jackson-datatype-jdk8#26]
in OptionalSerializer,

```java
    @Override
    public void acceptJsonFormatVisitor(JsonFormatVisitorWrapper visitor, JavaType typeHint) throws JsonMappingException
    {
        JsonSerializer<?> ser = _valueSerializer;
        if (ser == null) {
            ser = _findSerializer(visitor.getProvider(), _referredType, _property);
        }
        ser.acceptJsonFormatVisitor(visitor, _referredType);
    }
```

if _valueSerializer is null, when it finds the serializer it does not apply the _unwrapper if one exists.

Just need to add:

```java
                if (_unwrapper != null) {
                    ser = ser.unwrappingSerializer(_unwrapper);
                }
````

after the _findSerializer call

**Comment by cowtowncoder** [comment]
Seems like a simple fix, but in this case it would be good to have a unit test to verify behavior.
Especially since `OptionalSerializer` implementation was changed quite a bit between 2.6 and master, and I wouldn't wont to either break one by adding extra/unnecessary translation, or apply partial fix.

**Comment by osi** [comment]
no problem. I'm working on getting local builds working so I can send pull requests with tests

-- 
peter royal - (on the go)

**Comment by cowtowncoder** [comment]
Perfect. Should be easy enough to quash on my side then, include in 2.6.4.

**Comment by cowtowncoder** [comment]
Thank you for the patch; now merged for 2.6.4. I also applied the same patch to Guava `Optional` and JDK `AtomicReference`, which are similar referential types.
